src/main/java/moe/banana/jsonapi2/HasOne.java

```diff
@@ -19,6 +19,19 @@ public T get() throws ResourceNotFoundException {
         return (T) resource._doc.find(linkage);
     }
 
+    public T get(T defaultValue) {
+        try {
+            return get();
+        }
+        catch (ResourceNotFoundException e){
+            return defaultValue;
+        }
+    }
+
+    public boolean resourceExists(){
+        return resource._doc.contains(linkage);
+    }
+
```

Comment:
The name `resourceExists` is considered too verbose, `exists` or `isAccessible` can be better to read/write. Maybe we should rename the function with a more generally acceptable name so that `HasMany` can implement the same semantic with the same member function. `isResolvable` is one of the names I can come up with.